backend/app/rest/residents_routes.py

```diff
@@ -0,0 +1,66 @@
+from flask import Blueprint, current_app, jsonify, request
+from ..middlewares.auth import require_authorization_by_role
+from ..services.implementations.residents_service import ResidentsService
+import json
+
+residents_service = ResidentsService(current_app.logger)
+blueprint = Blueprint("residents", __name__, url_prefix="/residents")
+
+
+@blueprint.route("/", methods=["POST"], strict_slashes=False)
+@require_authorization_by_role({"Admin"})
+def add_resident():
+    """
+    Add a resident
+    """
+    resident = request.json
+    try:
+        created_resident = residents_service.add_resident(resident)
+        return jsonify(created_resident), 201
+    except Exception as e:
+        error_message = getattr(e, "message", None)
+        return jsonify({"error": (error_message if error_message else str(e))}), 500
+    
+#resident_id is the primary key in the table
+@blueprint.route("/<int:resident_id>", methods=["PUT"], strict_slashes=False)
+@require_authorization_by_role({"Admin"})
+def update_resident(resident_id):
+    """
+    Update an existing resident record based on the id 
+    """
+    updated_resident = request.json
+    try:
+     updated_resident = residents_service.update_resident(resident_id, updated_resident)
+     return jsonify({"message": "Resident record with id {resident_id} updated sucessfully".format(resident_id=resident_id)}), 201 
+    except Exception as e:
+        error_message = getattr(e, "message", None)
+        return jsonify({"error": (error_message if error_message else str(e))}), 500
+    
+@blueprint.route("/<int:resident_id>", methods=["DELETE"], strict_slashes=False)
+@require_authorization_by_role({"Admin"})
+def delete_resident(resident_id):
+    """
+    Delete a resident record based on id 
+    """
+    try:
+        residents_service.delete_resident(resident_id)
+        return jsonify({"message": "Resident with id {resident_id} deleted sucessfully".format(resident_id=resident_id)}), 201
+    except Exception as e:
+        error_message = getattr(e, "message", None)
+        return jsonify({"error": (error_message if error_message else str(e))}), 500     
+
+@blueprint.route("/", methods=["GET"], strict_slashes=False)
+def get_resident():
+    """
+    Get residents.
+    """ 
+    try:
+        resident_id = request.args.get("resident_id")
+        id = request.args.get("id")
```

Comment:
I feel like it makes more sense if the client could only filter residents by the resident ID only. IMO the `resident_id` is more meaningful than the `id` which is just some integer assigned by Postgres when the resident is added to the table. I feel like the nurses would know the residents by their resident ID and the code is easier to understand if you could only filter by one type of ID. Idk what the requirements of this PR were so lmk what you think

Reply:
I asked to include the id as a query param so that if we wanted to, we could also filter by id. For example, later we'll need to implement a page to store a table of residents, and we'd want the same functionality of pagination to retrieve an x amount of residents. With id as a param, we can retrieve lets say the first 50 ids and display those.

Reply:
wait no I'm dumb... we can just do a "SELECT * FROM ... LIMIT 50" so we don't need the id. You can remove it 👍 👍